src/main/java/org/kizombadev/markdownparser/SyntaxAnalyzer.java

```diff
@@ -46,103 +73,117 @@ public Syntax parse(ImmutableList<Token> tokens) {
+        MutableSyntax bigHeadline = new BigHeadline();
         stepTokenForward();
 
         handleLineContainer(bigHeadline);
         currentRoot.addChild(bigHeadline);
     }
 
-    private void handleSmallHeadlineLine(Syntax currentRoot) {
-        Syntax smallHeadline = new SmallHeadline();
+    private void handleSmallHeadlineLine(MutableSyntax currentRoot) {
+        MutableSyntax smallHeadline = new SmallHeadline();
         stepTokenForward();
 
 
         handleLineContainer(smallHeadline);
         currentRoot.addChild(smallHeadline);
     }
 
-    private void handleLineContainer(Syntax currentRoot) {
+    private void handleLineContainer(MutableSyntax currentRoot) {
 
         Token currentToken = currentToken();
 
         while (currentToken != null && !currentToken.equals(Token.NewLine)) {
 
-            if (currentToken.equals(Token.DoubleStar)) {
-                if (!stack.isEmpty() && stack.peek().equals(State.BOLD_BEGIN)) {
-                    stack.push(State.BOLD_END);
-                    return;
-                } else {
-                    handleBold(currentRoot);
-                }
+            if (currentToken.equals(Token.DoubleStar) && isBoldModusActive) {
+                break;
+            } else if (currentToken.equals(Token.DoubleStar)) {
+                handleBold(currentRoot);
+            } else if (currentToken.equals(Token.Star) && isItalicModusEnabled) {
+                break;
             } else if (currentToken.equals(Token.Star)) {
-                if (!stack.isEmpty() && stack.peek().equals(State.ITALIC_BEGIN)) {
-                    stack.push(State.ITALIC_END);
-                    return;
-                } else {
-                    handleItalic(currentRoot);
-                }
+                handleItalic(currentRoot);
             } else if (currentToken.isTextToken()) {
-                currentRoot.addChild(new TextSyntax());
+                currentRoot.addChild(new TextSyntax(currentToken.getTextValue()));
                 stepTokenForward();
             }
 
             currentToken = currentToken();
         }
     }
 
-    private void handleBold(Syntax currentRoot) {
+    private void handleBold(MutableSyntax currentRoot) {
+        isBoldModusActive = true;
         handleBeginEndToken(State.BOLD_BEGIN, State.BOLD_END, new BoldSyntax(), currentRoot);
-
+        isBoldModusActive = false;
     }
 
-    private void handleItalic(Syntax currentRoot) {
+    private void handleItalic(MutableSyntax currentRoot) {
+        isItalicModusEnabled = true;
         handleBeginEndToken(State.ITALIC_BEGIN, State.ITALIC_END, new ItalicSyntax(), currentRoot);
+        isItalicModusEnabled = false;
     }
 
-    private void handleBeginEndToken(State beginSate, State endState, Syntax newSyntaxElement, Syntax currentRoot) {
+    private void handleBeginEndToken(State beginSate, State endState, MutableSyntax newSyntaxElement, MutableSyntax currentRoot) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "beginSate". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)